app/src/main/java/com/zulip/android/activities/PhotoViewActivity.java

```diff
@@ -0,0 +1,225 @@
+        setContentView(R.layout.activity_photo_view);
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        getSupportActionBar().setHomeButtonEnabled(true);
+        final Intent intent = getIntent();
+        String url = intent.getStringExtra(Intent.EXTRA_TEXT);
+        getSupportActionBar().setTitle(url);
+        linkImage = (ImageView) findViewById(R.id.linkImageView);
+        progressBar = (ProgressBar) findViewById(R.id.progress);
+        GlideDrawableImageViewTarget imageViewPreview = new GlideDrawableImageViewTarget(linkImage);
+        Glide
+                .with(this)
+                .load(url)
+                .listener(new RequestListener<String, GlideDrawable>() {
+                    @Override
+                    public boolean onException(Exception e, String model, Target<GlideDrawable> target, boolean isFirstResource) {
+                        progressBar.setVisibility(View.GONE);
+                        if (!isNetworkAvailable()) {
+                            Toast.makeText(getApplicationContext(), R.string.toast_no_internet_connection, Toast.LENGTH_SHORT).show();
+                        } else {
+                            Toast.makeText(getApplicationContext(), R.string.toast_unable_to_load_image, Toast.LENGTH_SHORT).show();
+                        }
+                        return false;
+                    }
+
+                    @Override
+                    public boolean onResourceReady(GlideDrawable resource, String model, Target<GlideDrawable> target, boolean isFromMemoryCache, boolean isFirstResource) {
+                        progressBar.setVisibility(View.GONE);
+                        return false;
+                    }
+                })
+                .into(imageViewPreview);
+    }
+
+    private boolean isNetworkAvailable() {
+        ConnectivityManager connectivityManager
+                = (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkInfo activeNetworkInfo = connectivityManager.getActiveNetworkInfo();
+        return activeNetworkInfo != null && activeNetworkInfo.isConnected();
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.photoview_options, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        final Intent intent = getIntent();
+        final String url = intent.getStringExtra(Intent.EXTRA_TEXT);
+        switch (item.getItemId()) {
+            case R.id.open:
+                Intent i = new Intent(Intent.ACTION_VIEW);
+                i.setData(Uri.parse(url));
+                startActivity(i);
+                break;
+            case R.id.copy_link:
+                copyLink(url);
+                break;
+            case R.id.download:
+                downloadImage(url);
```

Comment:
@Sam1301 ofcourse It is. I still don't know how it successfully ran :O 
Thanks for pointing it out. I don't know how i missed it.